src/NavItem.js

```diff
@@ -28,7 +28,7 @@ class NavItem extends React.Component {
   }
 
   handleClick(e) {
-    if (this.props.onSelect) {
+    if (this.props.onSelect && this.props.onSelect.name !== 'bound handleCollapse') {
```

Comment:
agreed. my other thought was to pass down collapseOnSelect as a prop and check against that.